osf_tests/test_institution.py

```diff
@@ -103,3 +106,65 @@ def test_deactivated_institution_in_all_institutions(self):
         institution.deactivated = timezone.now()
         institution.save()
         assert institution in Institution.objects.get_all_institutions()
+
+    def test_deactivate_institution(self):
+        institution = InstitutionFactory()
+        with mock.patch.object(
+                institution,
+                '_send_deactivation_email',
+                return_value=None
+        ) as mock__send_deactivation_email:
+            institution.deactivate()
+            assert institution.deactivated is not None
+            assert mock__send_deactivation_email.called
+
+    def test_reactivate_institution(self):
+        institution = InstitutionFactory()
+        institution.deactivated = timezone.now()
+        institution.save()
+        institution.reactivate()
+        assert institution.deactivated is None
+
+    @mock.patch('website.mails.send_mail', side_effect=mails.send_mail)
+    def test_send_deactivation_email_call_count(self, mock_send_mail):
+        institution = InstitutionFactory()
+        user_1 = UserFactory()
+        user_1.affiliated_institutions.add(institution)
+        user_1.save()
+        user_2 = UserFactory()
+        user_2.affiliated_institutions.add(institution)
+        user_2.save()
+        institution._send_deactivation_email()
+        assert mock_send_mail.call_count == 2
+
+    def test_send_deactivation_email_call_args(self):
+        institution = InstitutionFactory()
+        user = UserFactory()
+        user.affiliated_institutions.add(institution)
+        user.save()
+        with mock.patch.object(mails, 'send_mail', return_value=None, side_effect=mails.send_mail) as mock_send_mail:
+            institution._send_deactivation_email()
+            forgot_password = 'forgotpassword' if settings.DOMAIN.endswith('/') else '/forgotpassword'
+            mock_send_mail.assert_called_with(
+                to_addr=user.username,
+                mail=mails.INSTITUTION_DEACTIVATION,
+                user=user,
+                forgot_password_link='{}{}'.format(settings.DOMAIN, forgot_password),
+                osf_support_email=settings.OSF_SUPPORT_EMAIL
+            )
```

Comment:
Here are the two tests with different mock style but both failed my local tests. I couldn't figure out why so I pushed the repo and see how GitHub Actions do. The issue is that `mock_send_mail` is not called at all in both cases.

Reply:
Here are the local `pytest` logs:

```
osf_tests/test_institution.py Cache table 'osf_cache_table' already exists.
[osf.models.institution]  INFO: Institution deactivation notification email has been sent to [2/2] users for [60da2d953649cc002fc7976d]
F[osf.models.institution]  INFO: Institution deactivation notification email has been sent to [1/1] users for [60da2d963649cc002fc79770]
F

================================================================================================================================================================================= FAILURES =================================================================================================================================================================================
______________________________________________________________________________________________________________________________________________________ TestInstitutionManager.test_send_deactivation_email_call_count ______________________________________________________________________________________________________________________________________________________
osf_tests/test_institution.py:138: in test_send_deactivation_email_call_count
    assert mock_send_mail.call_count == 2
E   AssertionError: assert 0 == 2
E    +  where 0 = <MagicMock name='send_mail' id='139974722629192'>.call_count
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
INFO     osf.models.institution:institution.py:194 Institution deactivation notification email has been sent to [2/2] users for [60da2d953649cc002fc7976d]
______________________________________________________________________________________________________________________________________________________ TestInstitutionManager.test_send_deactivation_email_call_args _______________________________________________________________________________________________________________________________________________________
osf_tests/test_institution.py:153: in test_send_deactivation_email_call_args
    osf_support_email=settings.OSF_SUPPORT_EMAIL
/usr/lib/python3.6/site-packages/mock/mock.py:925: in assert_called_with
    raise AssertionError('Expected call: %s\nNot called' % (expected,))
E   AssertionError: Expected call: send_mail(forgot_password_link='http://192.168.168.167:5000/forgotpassword', mail=<website.mails.mails.Mail object at 0x7f4e6affac88>, osf_support_email='fake-support@osf.io', to_addr='freddiemercury+5487715@cos.io', user=<OSFUser('freddiemercury+5487715@cos.io') with guid 't93fa'>)
E   Not called
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
INFO     osf.models.institution:institution.py:194 Institution deactivation notification email has been sent to [1/1] users for [60da2d963649cc002fc79770]
```

Reply:
Same [failures](https://github.com/CenterForOpenScience/osf.io/pull/9711/checks?check_run_id=2935651788#step:10:2763) in GA as expected.